WooCommerce/UITestsFoundation/Screens/ExternalAppScreen.swift

```diff
@@ -31,6 +31,6 @@ public final class ExternalAppScreen {
         safari.buttons["Go"].tap()
 
         // Tap on the universal link
-        if safari.staticTexts["TESTING LINKS"].exists { safari.links[universalLink].tap() }
+        if safari.links[universalLink].isHittable { safari.links[universalLink].tap() }
```

Comment:
Now, with a bit changed logic in the `if` condition, WDYT of either just going for the tap:

```suggestion
        safari.links[universalLink].tap()
```

or adding more logic (fake code):

```suggestion
        if safari.links[universalLink].isHittable { 
                safari.links[universalLink].tap()
        } else {
                FAIL_TEST(universal link was not found")
        }
```

Tapping a link is a critical part, and since there's no other failsafe (waiting for a link longer, reloading screen, etc), IMO it makes sense to fail the test as soon as we know it won't succeed.

Reply:
an earlier version of the code was without the `if` condition and it was pretty flaky without it. 

for the second part, i decided to implement that but using guard instead and to fail the test immediately if the condition is not met, updated in https://github.com/woocommerce/woocommerce-ios/pull/9493/commits/81468228306eba886c2c1529eaafb0762f67332e

this however won't solve the "odd" issue where the test ends up on the webview (i haven't seen that happening in CI so far so maybe not so much of an issue there)